### PR TITLE
CQ-4232394 : SRP clean up not working for ASRP setup

### PR DESCRIPTION
**Pull request by Vinod-holani**

### CODE REVIEW

File: bundles/cq-social-cleanup-srp/src/main/java/com/adobe/cq/social/cleanup_srp/impl/CleanupServlet.java [hunk 1]
```diff
@@ -115,15 +115,18 @@ private void cleanup(final ResourceResolver resolver, final int batchSize, final
...
-        rootResource.getResourceProvider().setConfig(config);
-        rootResource.getResourceProvider().delete(resolver, path);
-        try {
-            resolver.commit();
-        } catch (PersistenceException e) {
-            LOG.info("Could not persist deletion. Maybe there is nothing to delete", e);
-        }
-
+        // For SRPs that handle recursive delete
+        if (config.getAsiPath().equals("/content/usergenerated/asi/jcr")) {
```
Comment (neerava) [inline bundles/cq-social-cleanup-srp/src/main/java/com/adobe/cq/social/cleanup_srp/impl/CleanupServlet.java]: Please move this string in a static variable.

Comment (Vinod-holani) [reply bundles/cq-social-cleanup-srp/src/main/java/com/adobe/cq/social/cleanup_srp/impl/CleanupServlet.java]: Done. Please review

File: bundles/cq-social-cleanup-srp/src/main/java/com/adobe/cq/social/cleanup_srp/impl/CleanupServlet.java [hunk 2]
```diff
@@ -115,15 +116,18 @@ private void cleanup(final ResourceResolver resolver, final int batchSize, final
...
-        rootResource.getResourceProvider().setConfig(config);
-        rootResource.getResourceProvider().delete(resolver, path);
-        try {
-            resolver.commit();
-        } catch (PersistenceException e) {
-            LOG.info("Could not persist deletion. Maybe there is nothing to delete", e);
-        }
-
+        // For SRPs that handle recursive delete
+        if (config.getAsiPath().equals(JCR_ASI_PATH)) {
```
Comment (neerava) [inline bundles/cq-social-cleanup-srp/src/main/java/com/adobe/cq/social/cleanup_srp/impl/CleanupServlet.java:120]: Should it be equals() or startsWith()?

Comment (Vinod-holani) [reply bundles/cq-social-cleanup-srp/src/main/java/com/adobe/cq/social/cleanup_srp/impl/CleanupServlet.java:120]: It is exact ASI path to be compared so used equals(). startWith() will also work for the same.
Should it be modified?

**Pull request merged**